proxy/proxy.go

```diff
@@ -848,6 +848,9 @@ func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Co
 	if endpoint != nil {
 		endpoint.Metrics.IncInflightRequest()
 		defer endpoint.Metrics.DecInflightRequest()
+
+		p.registry.IncInflightRequest(endpoint.Host)
+		defer p.registry.DecInflightRequest(endpoint.Host)
```

Comment:
Can it happen that we increment and decrement on different instances of metrics here?